lib/account.py

```diff
@@ -1,74 +1,103 @@
+        I'm not sure if this where risk management and/or portfolio balancing
+        should be done though... likely not, but it shouldn't be done in the
+        Runner either...
     """
 
-    can_withdraw = None
-    can_trade = None
-    can_deposit = None
-    balances = {}
-
-    def __init__(self, payload):
-        self.can_withdraw = payload["canWithdraw"]
-        self.can_trade = payload["canTrade"]
-        self.can_deposit = payload["canDeposit"]
-
-        for asset in payload["balances"]:
-            self.balances[asset["asset"]] = (
-                float(asset["free"]), float(asset["locked"]))
-
-    def __call__(self, payload):
-        """
-            For simplicity Account is callable.
-        """
-
-        action = {
-            'outboundAccountInfo': self.account_update,
-            'outboundAccountPosition': self.position_update,
-            'balanceUpdate': self.balance_update,
-            'executionReport': self.order_report
-        }.get(payload["e"],
-              lambda p: print("unhandled account event ", p["e"]))
-
-        action(payload)
-
-    def account_update(self, payload):
-        self.can_withdraw = payload["W"]
-        self.can_trade = payload["T"]
-        self.can_deposit = payload["D"]
-
-        for asset in payload["B"]:
-            self.balances[asset["a"]] = (float(asset["f"]), float(asset["l"]))
-
-    def position_update(self, payload):
-        for asset in payload["B"]:
-            self.balances[asset["a"]] = (float(asset["f"]), float(asset["l"]))
-
-    def balance_update(self, payload):
-        self.balances[payload["a"]] = (
-            self.balances[payload["a"]][0] + float(payload["d"]),
-            self.balances[payload["a"]][1]
-        )
-
-    def order_report(self, payload):
-        # Not entirely sure what, if anything other than logging, we should do
-        # here. After all, actual account updates/state are already handled.
+    def __call__(self, update_type: str, updated_props: Update):
+        """Sets account state - i.e. balance and positions."""
+
+        # This required python > 3.10.*; alas there's a bug with pip
+        # on this version. So this will likely need to be refactored
+        # to use simple if/else comparisons until pip 21.2.3 is released.
+        # @see https://github.com/pypa/pip/pull/10252
+        # @todo
+        match update_type:
+            case UpdateType.BALANCE:
+                # @todo Eh, look at whatever fuckery is involved.
+                # surely the "Locked" balance needs updating to..?!
+                asset = updated_props["Asset"]
+                self.status.Positions[asset]["Free"] += updated_props["Update"]
+            case UpdateType.ACCOUNT:
+                self.status = updated_props
+            case UpdateType.POSITION:
+                self.status["Positions"] = updated_props
+
+
+    def initial_state(self, status: AccountStatus):
+        self.status = status
+
+    def purchase(self, symbol: str):
```

Comment:
This will wrap around a `Queue` as part of the *Actor* implementation; in which case a `receive or `handle` method may make more sense - as an interface for all market interactions